go/obscuronode/enclave/enclave.go

```diff
@@ -362,14 +366,20 @@ func (e *enclaveImpl) produceRollup(b *types.Block, bs *obscurocore.BlockState)
 	// always process deposits last, either on top of the rollup produced speculatively or the newly created rollup
 	// process deposits from the proof of the parent to the current block (which is the proof of the new rollup)
 	proof := e.blockResolver.Proof(headRollup)
-	depositTxs := processDeposits(proof, b, e.blockResolver, e.txHandler)
+	depositTxs := extractDeposits(proof, b, e.blockResolver, e.txHandler)
 	executeTransactions(depositTxs, newRollupState, newRollupHeader)
 
 	// Create a new rollup based on the proof of inclusion of the previous, including all new transactions
-	r := obscurocore.NewRollupFromHeader(newRollupHeader, b.Hash(), newRollupTxs, obscurocommon.GenerateNonce(), newRollupState.StateRoot())
+	rootHash, err := newRollupState.Commit(true)
+	if err != nil {
+		return nil
+	}
+	// dump := newRollupState.Dump(&state.DumpConfig{})
+	// log.Log(fmt.Sprintf(">   Agg%d: State:%s", obscurocommon.ShortAddress(e.node), dump))
+	r := obscurocore.NewRollupFromHeader(newRollupHeader, b.Hash(), newRollupTxs, obscurocommon.GenerateNonce(), rootHash)
```

Comment:
Should these comments be removed ?

Reply:
dumping is super useful fo debugging, I'll figure somethign in the next iteration to add it as a trace or something